### PR TITLE
allow discoveryAddress to be set via proxyConfig for VMs with revision

### PR DESCRIPTION
**Pull request by stevenctl**
similar to #34060 – need to do more investigation on the safety and triple check the test does what I think it does 

* This should make `CA_ADDR`, `PILOT_ADDRESS` and `ISTIO_PILOT_PORT` vars still work in a backward compatible way, but avoid overriding with the default if the user didn't customize the envs. 
* The workload command now properly sets the port for the discovery address
* The workload command sets the CA_ADDR to match the discovery address (if rev is specified)
* The tests actually copy `sidecar.env` 
* The tests actually deploy a VM
* The test send traffic from the VM to make sure it connected to istiod properly
* The tests deploy a VirtualService pointing to the revisioned version of the eastwest gateway (this may need some cleanup to match the specific SNI)

### CODE REVIEW

File: tools/packaging/common/istio-start.sh [hunk 1]
```diff
@@ -116,11 +117,18 @@ ISTIO_AGENT_FLAGS=${ISTIO_AGENT_FLAGS:-}
...
-export PROXY_CONFIG=${PROXY_CONFIG:-"
+DEFAULT_PROXY_CONFIG="
 serviceCluster: $SVC
 controlPlaneAuthPolicy: ${CONTROL_PLANE_AUTH_POLICY}
-discoveryAddress: ${PILOT_ADDRESS}
-"}
+"
+if [ -n "${CUSTOM_PILOT_ADDRESS}" ]; then
+  PROXY_CONFIG="$PROXY_CONFIG
+  discoveryAddress: ${CUSTOM_PILOT_ADDRESS}
```
Comment (howardjohn) [inline tools/packaging/common/istio-start.sh:126]:
```suggestion
discoveryAddress: ${CUSTOM_PILOT_ADDRESS}
```
spacing is suspicious here?

File: tests/integration/pilot/revisions/revisions_test.go [hunk 2]
```diff
@@ -95,9 +95,10 @@ func TestMultiRevision(t *testing.T) {
 				}).
 				// tests bootstrap
 				WithConfig(echo.Config{
-					Service:   "vm",
-					Namespace: canary,
-					Ports:     []echo.Port{},
+					Service:    "vm",
+					Namespace:  canary,
+					DeployAsVM: true,
```
Comment (howardjohn) [inline tests/integration/pilot/revisions/revisions_test.go:100]: oof

**Review by howardjohn — APPROVED** [review]

File: tests/integration/pilot/revisions/revisions_test.go (hunk 2 above)
Comment (stevenctl) [reply tests/integration/pilot/revisions/revisions_test.go:100]: I still feel like the test should fail. We use https://github.com/istio/istio/blob/master/samples/multicluster/expose-istiod.yaml which doesn't use the revisioned version. I'm running manually to verify

File: samples/multicluster/expose-istiod-rev.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - istiod-gateway
+  tls:
+  - match:
+    - port: 15012
+      sniHosts:
+      - "*"
```
Comment (stevenctl) [inline samples/multicluster/expose-istiod-rev.yaml:39]: Assuming  this should be `istiod-rev.*`? @howardjohn

Comment (howardjohn) [reply samples/multicluster/expose-istiod-rev.yaml:39]: I don't think prefix match is supported

**Review by Monkeyanator — APPROVED** [review]

**Comment by xulingqing** [comment]
@istio/wg-user-experience-maintainers could u take a look? Thanks!

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**